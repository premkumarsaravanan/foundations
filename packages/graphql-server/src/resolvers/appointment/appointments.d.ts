export type AppointmentContact = {
  id: string

  name: string

  homePhone: string

  workPhone: string

  mobilePhone: string

  email: string
}

export type AppointmentAttendee = {
  id: string

  type: string

  contacts: [AppointmentContact]
}

export type AppointmentFollowUp = {
  due: string

  responseId: string

  notes: string
}

export type Appointment = {
  id: string

  created: string

  modified: string

  start: string

  end: string

  typeId: string

  description: string

  recurring: boolean

  cancelled: boolean

  followUp: AppointmentFollowUp

  propertyId: string

  organiserId: string

  negotiatorIds: string[]

  officeIds: string[]

  attendee: AppointmentAttendee

  accompanied: boolean

  negotiatorConfirmed: boolean

  attendeeConfirmed: boolean

  propertyConfirmed: boolean

  metadata: {
    [name: string]: any
  }

  _eTag: string
  _links: {
    [name: string]: any
  }
  _embedded: {
    [name: string]: any
  }
}

export type UpdateAppointmentInputAttendee = {
  id: string

  type: string

  confirmed: boolean
}

export type CreateAppointmentInputAttendee = {
  id: string
  type: string
}

export type CreateAppointmentInputRecurrence = {
  interval: number

  type: string

  until: string
}

export type UpdateAppointmentInputRecurrence = {
  type: string

  interval: number

  until: string
}

export type UpdateAppointmentInputFollowUp = {
  responseId: string
  notes: string
}

export type SearchAppointmentArgs = {
  start: string
  end: string
  followUpOn: string
  typeId: string
  description: string
  organiserId: string
  negotiatorIds: string[]
  officeIds: string[]
  attendee: CreateAppointmentInputAttendee
  propertyId: string
  accompanied: boolean
  negotiatorConfirmed: boolean
  attendeeConfirmed: boolean
  propertyConfirmed: boolean
  recurrence: CreateAppointmentInputRecurrence
  metadata: {
    [name: string]: any
  }
}

export type CreateAppointmentArgs = {
  start: string
  end: string
  followUpOn: string
  typeId: string
  description: string
  organiserId: string
  negotiatorIds: string[]
  officeIds: string[]
  attendee: CreateAppointmentInputAttendee
  propertyId: string
  accompanied: boolean
  negotiatorConfirmed: boolean
  attendeeConfirmed: boolean
  propertyConfirmed: boolean
  recurrence: CreateAppointmentInputRecurrence
  metadata: {
    [name: string]: any
  }
}

export type UpdateAppointmentArgs = {
  id: string
  start: string
  end: string
  followUpOn: string
  typeId: string
  description: string
  propertyId: string
  organiserId: string
  cancelled: boolean
  negotiatorIds: string[]
  officeIds: string[]
  attendee: UpdateAppointmentInputAttendee
  accompanied: boolean
  negotiatorConfirmed: boolean
  attendeeConfirmed: boolean
  propertyConfirmed: boolean
  followUp: UpdateAppointmentInputFollowUp
  recurrence: UpdateAppointmentInputRecurrence
  metadata: {
    [name: string]: any
  }
  _eTag: string
}

export type GetAppointmentByIdArgs = {
  id: string
}

export type GetAppointmentsResult = {
  pageNumber: number
  pageSize: number
  pageCount: number
  totalCount: number
  readonly _links?: {
    [name: string]: {
      href?: string
    }
  }
  _embedded: [Appointment]
}

export type GetAppointmentsArgs = {
  pageSize?: number
  pageNumber?: number
  sortBy?: string
  id?: string[]
  typeId?: string[]
  negotiatorId?: string[]
  officeId?: string[]
  propertyId?: string[]
  start: string
  end: string
  includeCancelled?: boolean
  includeUnconfirmed?: boolean
}
