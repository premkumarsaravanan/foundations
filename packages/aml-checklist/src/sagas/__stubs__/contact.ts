export const contact = {
  id: 'MKC13000122',
  created: '2017-01-10T11:33:37.0000000Z',
  modified: '2020-01-21T14:04:50.0000000Z',
  title: 'Msss',
  forename: 'Holly',
  surname: 'Rees',
  dateOfBirth: '1986-06-30',
  active: true,
  marketingConsent: 'notAsked',
  source: {
    id: '',
    type: ''
  },
  homePhone: '01632 968608',
  workPhone: '020 7946 8608',
  mobilePhone: '07700 908608',
  email: 'hrees927@rpsfiction.net',
  primaryAddress: {
    type: 'home',
    buildingName: 'Larch Cottage',
    buildingNumber: '12',
    line1: 'High Street North',
    line2: 'Stewkley',
    line3: 'Leighton Buzzard',
    line4: 'Buckinghamshire',
    postcode: 'LU7 0EP',
    countryId: 'GB'
  },
  secondaryAddress: {
    type: 'previous',
    buildingName: 'Larch Cottage',
    buildingNumber: '7',
    line1: 'Ledburn',
    line2: 'Leighton Buzzard',
    line3: 'Buckinghamshire',
    line4: '',
    postcode: 'LU7 0PX',
    countryId: ''
  },
  officeIds: ['MKC'],
  negotiatorIds: ['JPB'],
  metadata: {
    addresses: [
      {
        documentImage:
          'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/home-12-Larch Cottage-LU7 0EP.png',
        year: '2',
        month: '2',
        documentType: 'Bank / Building Society / Statement'
      },
      {
        documentImage:
          'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/previous-7-Larch Cottage-LU7 0PX(1).jpg',
        year: '3',
        month: '1',
        documentType: 'Credit Card Statements from main provider'
      }
    ],
    declarationRisk: {
      reason: 'Test',
      type: 'Simplified',
      riskAssessmentForm:
        'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/riskAssessment-Simplified-Test(1).png'
    }
  },
  _links: {
    self: {
      href: '/contacts/MKC13000122'
    },
    documents: {
      href: '/documents/?OwnerType=contact&OwnerId=MKC13000122'
    },
    identityChecks: {
      href: '/identityChecks/?ContactId=MKC13000122'
    }
  }
}

export const idCheck = {
  id: 'RPT19000083',
  contactId: 'MKC13000122',
  created: '0001-01-01T00:00:00',
  modified: '2020-01-21T03:39:53',
  checkDate: '0001-01-01T00:00:00',
  status: 'pass',
  negotiatorId: 'LJW',
  document1: {
    typeId: 'PP',
    expiry: '2020-07-03T23:00:00',
    details: '123123'
  },
  document2: {
    typeId: 'BB',
    expiry: '2020-01-12T00:00:00',
    details: 'ID Reference'
  },
  metadata: {
    primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/AYL19000001-testst.png',
    secondaryIdUrl: 'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/MKC13000122-ID Reference.png'
  },
  _links: {
    self: {
      href: '/identityChecks/RPT19000083'
    },
    contact: {
      href: '/contacts/MKC13000122'
    },
    identityDocument1Type: {
      href: '/configuration/identityDocumentTypes/PP'
    },
    identityDocument2Type: {
      href: '/configuration/identityDocumentTypes/BB'
    }
  }
}
