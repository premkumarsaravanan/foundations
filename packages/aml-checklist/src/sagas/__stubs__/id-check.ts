export const idCheck = {
  id: 'RPT19000104',
  contactId: 'AYL19000001',
  created: '0001-01-01T00:00:00',
  modified: '2019-12-13T05:41:45',
  checkDate: '0001-01-01T00:00:00',
  status: 'pass',
  negotiatorId: 'LJW',
  document1: {
    typeId: 'TX',
    expiry: '2020-02-07T00:00:00',
    details: 'Hshs'
  },
  document2: {
    typeId: 'CI',
    expiry: '2019-12-21T00:00:00',
    details: 'a'
  },
  metadata: {
    primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/AYL19000001-testst.png'
  },
  _links: {
    self: {
      href: '/identityChecks/RPT19000104'
    },
    contact: {
      href: '/contacts/AYL19000001'
    },
    identityDocument1Type: {
      href: '/configuration/identityDocumentTypes/TX'
    },
    identityDocument2Type: {
      href: '/configuration/identityDocumentTypes/CI'
    }
  }
}
