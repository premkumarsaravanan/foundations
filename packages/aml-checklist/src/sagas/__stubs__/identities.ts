export const identities = {
  _embedded: [
    {
      id: 'RPT19000114',
      contactId: 'MKC19000013',
      created: '0001-01-01T00:00:00',
      modified: '2019-11-01T15:28:09',
      checkDate: '0001-01-01T00:00:00',
      status: 'pass',
      negotiatorId: 'LJW',
      document1: {
        typeId: 'DL',
        expiry: '2019-11-14T00:00:00',
        details: 'adwqr'
      },
      metadata: {
        primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/MKC19000013-adwqr.png'
      },
      _links: {
        self: {
          href: '/identityChecks/RPT19000114'
        },
        contact: {
          href: '/contacts/MKC19000013'
        },
        identityDocument1Type: {
          href: '/configuration/identityDocumentTypes/DL'
        }
      }
    }
  ],
  pageNumber: 1,
  pageSize: 50,
  pageCount: 1,
  totalCount: 1,
  _links: {
    self: {
      href:
        '/identityChecks/?PageNumber=1&PageSize=50&ContactId=MKC19000013&ContactId=BED16000182&ContactId=MKC16000060&ContactId=BED16000099&ContactId=MKC16000011&ContactId=MKC15000909&ContactId=BUC15000679&ContactId=BED15002235&ContactId=BED15002208&ContactId=BED15001889'
    },
    first: {
      href:
        '/identityChecks/?PageNumber=1&PageSize=50&ContactId=MKC19000013&ContactId=BED16000182&ContactId=MKC16000060&ContactId=BED16000099&ContactId=MKC16000011&ContactId=MKC15000909&ContactId=BUC15000679&ContactId=BED15002235&ContactId=BED15002208&ContactId=BED15001889'
    }
  }
}
