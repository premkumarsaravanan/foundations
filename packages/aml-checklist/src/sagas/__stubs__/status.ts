import { SectionsStatus } from '@/reducers/checklist-detail'

export const sectionsStatus: SectionsStatus = {
  profile: true,
  primaryId: false,
  secondaryId: true,
  declarationRisk: false,
  addresses: false
  // pepSearch: false,
  // experian: true
}
