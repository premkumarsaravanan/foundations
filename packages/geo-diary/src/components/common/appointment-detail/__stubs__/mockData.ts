export const attendees = {
  negotiator: {
    id: 'Negotiator',
    type: 'negotiator',
    name: 'Negotiator',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Negotiator',
        detail: 'Negotiator'
      }
    ]
  },
  office: {
    id: 'Office',
    type: 'office',
    name: 'Office',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Office',
        detail: 'Office'
      }
    ]
  },
  landlord: {
    id: 'Landlord',
    type: 'landlord',
    name: 'Landlord',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Landlord',
        detail: 'Landlord'
      }
    ]
  },
  contact: {
    id: 'Contact',
    type: 'contact',
    name: 'Contact',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Contact',
        detail: 'Contact'
      }
    ]
  },
  applicant: {
    id: 'Applicant',
    type: 'applicant',
    name: 'Applicant',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Applicant',
        detail: 'Applicant'
      }
    ]
  },
  tenant: {
    id: 'Tenant',
    type: 'tenant',
    name: 'Tenant',
    confirmed: true,
    communicationDetails: [
      {
        label: 'Tenant',
        detail: 'Tenant'
      }
    ]
  }
}
