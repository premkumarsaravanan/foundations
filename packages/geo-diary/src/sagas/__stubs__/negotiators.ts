import { NegotiatorModel } from '@reapit/foundations-ts-definitions'

export const negotiatorStub: NegotiatorModel = {
  id: 'string',
  created: '2020-01-21T19:35:42.241Z',
  modified: '2020-01-21T19:35:42.241Z',
  name: 'string',
  jobTitle: 'string',
  active: true,
  officeId: 'string',
  workPhone: 'string',
  mobilePhone: 'string',
  email: 'string',
  metadata: {
    additionalProp1: {},
    additionalProp2: {},
    additionalProp3: {}
  },
  _links: {
    additionalProp1: {
      href: 'string'
    },
    additionalProp2: {
      href: 'string'
    },
    additionalProp3: {
      href: 'string'
    }
  },
  _embedded: {
    additionalProp1: {},
    additionalProp2: {},
    additionalProp3: {}
  }
}
