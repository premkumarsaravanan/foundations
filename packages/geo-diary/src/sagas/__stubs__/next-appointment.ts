import { NextAppointmentState } from '@/reducers/next-appointment'

export const nextAppointmentDataStub: NextAppointmentState = {
  data: {
    id: 'jk287ch',
    attendeeWithMobile: undefined,
    currentNegotiator: undefined,
    durationValue: 900,
    durationText: '1h 30 mins',
    distanceValue: 100,
    distanceText: '1 mile'
  }
}
