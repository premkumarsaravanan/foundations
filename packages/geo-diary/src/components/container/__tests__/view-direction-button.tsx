import * as React from 'react'
import { ViewDirectionButton, mapDispatchToProps } from '../view-direction-button'
import { shallow } from 'enzyme'
import { appointmentDataStub } from '@/sagas/__stubs__/appointment'

describe('ViewDirectionButton', () => {
  it('Should match snapshot', () => {
    const mockProps = {
      handleOnClick: jest.fn()
    }
    const wrapper = shallow(<ViewDirectionButton {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call handleOnClick when clicked', () => {
    const mockProps = {
      handleOnClick: jest.fn()
    }
    const wrapper = shallow(<ViewDirectionButton {...mockProps} />)
    wrapper.simulate('click')
    expect(mockProps.handleOnClick).toBeCalled()
  })

  describe('mapDispatchToProps', () => {
    const mockedDispatch = jest.fn()
    const mockedOwnProps = {
      appointment: appointmentDataStub
    }

    const { handleOnClick } = mapDispatchToProps(mockedDispatch, mockedOwnProps)
    it('run handleOnClick successfully', () => {
      handleOnClick()
      expect(mockedDispatch).toBeCalledTimes(3)
    })
  })
})
