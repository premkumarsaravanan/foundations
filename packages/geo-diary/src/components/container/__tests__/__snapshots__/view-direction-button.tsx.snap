// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDirectionButton Should match snapshot 1`] = `
<Button
  className="is-centered"
  onClick={[MockFunction]}
  type="button"
  variant="info"
>
  Directions
</Button>
`;
